### PR TITLE
fix: healthcheck for docker containers

### LINKED ISSUE #2841: Dragonfly Docker image v1.16.0 healtcheck is broken
**Describe the bug**
After updating (freshly pulling) the latest image (v1.16.0), docker container crashes (during start) with the following log.

```
I20240404 10:51:11.817468     1 init.cc:70] dragonfly running in opt mode.
I20240404 10:51:11.817705     1 dfly_main.cc:641] Starting dragonfly df-v1.16.0-8bd35754de9ae49908369961634dad0b7fbea878
* Logs will be written to the first available of the following paths:
/tmp/dragonfly.*
./dragonfly.*
* For the available flags type dragonfly [--help | --helpfull]
* Documentation can be found at: https://www.dragonflydb.io/docs
W20240404 10:51:11.818120     1 dfly_main.cc:680] SWAP is enabled. Consider disabling it when running Dragonfly.
I20240404 10:51:11.818161     1 dfly_main.cc:685] maxmemory has not been specified. Deciding myself....
I20240404 10:51:11.818197     1 dfly_main.cc:694] Found 460.68GiB available memory. Setting maxmemory to 368.54GiB
W20240404 10:51:11.818303     1 dfly_main.cc:368] Weird error 1 switching to epoll
I20240404 10:51:11.908849     1 proactor_pool.cc:146] Running 48 io threads
I20240404 10:51:11.980602     1 server_family.cc:713] Host OS: Linux 5.15.0-101-generic x86_64 with 48 threads
I20240404 10:51:12.010690     1 snapshot_storage.cc:108] Load snapshot: Searching for snapshot in directory: "/data"
W20240404 10:51:12.010876     1 server_family.cc:806] Load snapshot: No snapshot found
I20240404 10:51:12.038388     9 listener_interface.cc:101] sock[99] AcceptServer - listening on port 6379
I20240404 10:52:44.101867     8 accept_server.cc:24] Exiting on signal Terminated
I20240404 10:52:44.103591     9 listener_interface.cc:201] Listener stopped for port 6379
I20240404 10:52:44.149309    13 save_stages_controller.cc:321] Saving "dump-2024-04-04T10:52:44-summary.dfs" finished after 0 us
```
The workaround was pinning the image back to v1.15.1, then it starts normal again
Not sure if related to #2739

**To Reproduce**
Start docker compose with this config
```yaml
version: '3.9'
services:

  dragonfly:
    image: docker.dragonflydb.io/dragonflydb/dragonfly #latest is v1.16.0 at this time
    # image: docker.dragonflydb.io/dragonflydb/dragonfly:v1.15.1 #fixes the problem
    restart: unless-stopped
    ulimits:
      memlock: -1
    volumes:
      - pretalx-redis:/data

  # Other services...

volumes:
  pretalx-redis:
  # Other volumes not related to dragonfly ...
```

**Expected behavior**
Container won't crash

**Environment (please complete the following information):**
 - Containerized?: Docker Compose (via Portainer)
 - Dragonfly Version: v1.16.0

### PR DESCRIPTION
**Pull request by romange**
Fixes #2841

### CODE REVIEW

File: tools/docker/healthcheck.sh [hunk 1]
```diff
@@ -4,7 +4,8 @@ HOST="localhost"
 PORT=$HEALTHCHECK_PORT
 
 if [ -z "$HEALTHCHECK_PORT" ]; then
-    PORT=$(netstat -tuln | grep -oE ':[0-9]+' | grep -oE '[0-9]+' | tail -n 1)
+    # check all the TCP listening sockets, filter the dragonfly process, and fetch the port
+    PORT=$(netstat -tlnp | grep "1/dragonfly" | grep -oE ':[0-9]+' | cut -c2-)
```
Comment (Abhra303) [inline tools/docker/healthcheck.sh:8]: It returns two ports if admin port enabled - 
```
6379
9999
```
Choosing one of them would fix it (piping it with `tail -n 1`)

Comment (romange) [reply tools/docker/healthcheck.sh:8]: good catch!

**Review by Abhra303 — APPROVED** [review]
LGTM

**Comment by romange** [comment]
@chakaz  pls approve, Abhra already reviewed.

**Review by ashotland — APPROVED** [review]

**Comment by chakaz** [comment]
LGTM too :)

**Pull request merged**